src/include/security/levels.rb

```
@@ -65,127 +68,15 @@ def initialize_security_levels(include_target)
-            "PASS_MIN_DAYS"                             => "1",
-            "PASS_MIN_LEN"                              => "6",
-            "PASS_WARN_AGE"                             => "14",
-            "PERMISSION_SECURITY"                       => "secure",
-            "RUN_UPDATEDB_AS"                           => "nobody",
-            "UID_MAX"                                   => "60000",
-            "UID_MIN"                                   => "1000",
-            "SYS_UID_MAX"                               => "499",
-            "SYS_UID_MIN"                               => "100",
-            "SYS_GID_MAX"                               => "499",
-            "SYS_GID_MIN"                               => "100",
-            "USERADD_CMD"                               => "/usr/sbin/useradd.local",
-            "USERDEL_PRECMD"                            => "/usr/sbin/userdel-pre.local",
-            "USERDEL_POSTCMD"                           => "/usr/sbin/userdel-post.local",
-            "PASSWD_REMEMBER_HISTORY"                   => "0",
-            "SYSTOHC"                                   => "yes",
-            "SYSLOG_ON_NO_ERROR"                        => "yes",
-            "DISPLAYMANAGER_ROOT_LOGIN_REMOTE"          => "no",
-            "DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN" => "no",
-            "SMTPD_LISTEN_REMOTE"                       => "no",
-            "DISABLE_STOP_ON_REMOVAL"                   => "no",
-            "DISABLE_RESTART_ON_UPDATE"                 => "no",
-            "net.ipv4.tcp_syncookies"                   => "1",
-            "net.ipv4.ip_forward"                       => "0",
-            "net.ipv6.conf.all.forwarding"              => "0"
-          }
-        } 
+      @Levels = {}
+      @LevelsNames.keys.each do |level|
+        lfile = Directory.find_data_file("security/#{level.downcase}.yml")
+        if lfile
+          @Levels[level] = YAML.load_file(lfile) rescue {}
```

this is really not nice as it hide possible errors in file. Consider that you modify file for testing and use wrong syntax and result is that whole file is empty? At least some logging would be nice.